api/ipfsproxy/config.go

```diff
@@ -116,26 +128,49 @@ func (cfg *Config) LoadJSON(raw []byte) error {
 		return err
 	}
 
-	cfg.Default()
+	if jcfg.ListenMultiaddress == "" {
+		jcfg.ListenMultiaddress = jcfg.ProxyListenMultiaddress
+	}
+
+	if jcfg.ReadTimeout == "" {
+		jcfg.ReadTimeout = jcfg.ProxyReadTimeout
+	}
+
+	if jcfg.ReadHeaderTimeout == "" {
+		jcfg.ReadHeaderTimeout = jcfg.ProxyReadHeaderTimeout
+	}
+
+	if jcfg.WriteTimeout == "" {
+		jcfg.WriteTimeout = jcfg.ProxyWriteTimeout
+	}
+
+	if jcfg.IdleTimeout == "" {
+		jcfg.IdleTimeout = jcfg.ProxyIdleTimeout
+	}
+
+	err = cfg.Default()
+	if err != nil {
+		return fmt.Errorf("error setting config to default values: %s", err)
+	}
```

Comment:
After setting the defaults, add:
```go
	// override json config with env var
	err = envconfig.Process(envConfigKey, jcfg)
	if err != nil {
		return err
	}
```

And add `const envConfigKey = "cluster_ipfsproxy"` after the `configKey` const at the top of the file.

Reply:
hmm, I don't see any such environment variable being used at the moment. Are you suggesting we start using them?

Reply:
I have been attempting to make setting all configuration values to be **possible** by environment variables, see the discussion in #575. This has mainly stemmed from the pain points I have and others have hit when deploying ipfs and cluster in cloud-native ways.

Reply:
I could not find any documentation about all these env variables. If it doesn't exist, let's create an issue for that.

Reply:
Hope that is in the right repo https://github.com/ipfs/ipfs-cluster-website/issues/47